absl/container/internal/hash_function_defaults_test.cc

```diff
@@ -309,17 +311,33 @@ TYPED_TEST(EqPointer, Works) {
   const int* cptr = ptr;
   std::unique_ptr<int, NoDeleter> uptr(ptr);
   std::unique_ptr<const int, NoDeleter> cuptr(ptr);
+  auto saptr = std::shared_ptr<int[]>(new int[1]);
```

Comment:
As far as I can tell, this isn't legal in C++14, even though most compiler/library combinations accept it. libc++ seems to reject it.

https://godbolt.org/z/qK61vPj65

I think it has something to do with `new int[1]` decaying to `int*`, which would lead to the wrong choice of a deleter.